samza-core/src/main/java/org/apache/samza/zk/ScheduleAfterDebounceTime.java

```diff
@@ -72,13 +84,29 @@ synchronized public void scheduleAfterDebounceTime(String actionName, long debou
       futureHandles.remove(actionName);
     }
     // schedule a new task
-    sf = scheduledExecutorService.schedule(runnable, debounceTimeMs, TimeUnit.MILLISECONDS);
-    LOGGER.info("DEBOUNCE: scheduled " + actionName + " in " + debounceTimeMs);
+    sf = scheduledExecutorService.schedule(() -> {
+        try {
+          runnable.run();
+          LOGGER.debug(actionName + " completed successfully.");
+        } catch (Exception e) {
```

Comment:
sorry, my bad, I used the wrong class name. It's Error, not RuntimeException (https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html), I think we want to catch those too. Otherwise the thread might die without notifying the listener. So it is safer to catch all the throwable and pass on to the listener.